rosidl_typesupport_c/resource/rosidl_typesupport_c__visibility_control.h.in

```diff
@@ -1,43 +0,0 @@
-// generated from
-// rosidl_typesupport_c/resource/rosidl_typesupport_c__visibility_control.h.in
-// generated code does not contain a copyright notice
-
-#ifndef @PROJECT_NAME_UPPER@__MSG__ROSIDL_TYPESUPPORT_C__VISIBILITY_CONTROL_H_
-#define @PROJECT_NAME_UPPER@__MSG__ROSIDL_TYPESUPPORT_C__VISIBILITY_CONTROL_H_
-
-#ifdef __cplusplus
-extern "C"
-{
-#endif
-
-// This logic was borrowed (then namespaced) from the examples on the gcc wiki:
-//     https://gcc.gnu.org/wiki/Visibility
-
-#if defined _WIN32 || defined __CYGWIN__
-  #ifdef __GNUC__
-    #define ROSIDL_TYPESUPPORT_C_EXPORT_@PROJECT_NAME@ __attribute__ ((dllexport))
-    #define ROSIDL_TYPESUPPORT_C_IMPORT_@PROJECT_NAME@ __attribute__ ((dllimport))
-  #else
-    #define ROSIDL_TYPESUPPORT_C_EXPORT_@PROJECT_NAME@ __declspec(dllexport)
-    #define ROSIDL_TYPESUPPORT_C_IMPORT_@PROJECT_NAME@ __declspec(dllimport)
-  #endif
-  #ifdef ROSIDL_TYPESUPPORT_C_BUILDING_DLL_@PROJECT_NAME@
-    #define ROSIDL_TYPESUPPORT_C_PUBLIC_@PROJECT_NAME@ ROSIDL_TYPESUPPORT_C_EXPORT_@PROJECT_NAME@
-  #else
-    #define ROSIDL_TYPESUPPORT_C_PUBLIC_@PROJECT_NAME@ ROSIDL_TYPESUPPORT_C_IMPORT_@PROJECT_NAME@
-  #endif
-#else
-  #define ROSIDL_TYPESUPPORT_C_EXPORT_@PROJECT_NAME@ __attribute__ ((visibility("default")))
```

Comment:
It's not that they're not necessary, but that there's already a visibility file for this symbol created by `rosidl_generator_c`. This generator creates the `.cpp` file which has the symbol in a cpp file declared with the export macro from this file

https://github.com/ros2/rosidl_typesupport/blob/97bc0a7716f7346dc0a34eeaba2e9b658f59beb5/rosidl_typesupport_c/resource/msg__type_support.cpp.em#L128-L130

But the header included by downstream packages is generated by `rosidl_generator_c`, and that uses a different visibility header file also generated by `rosidl_generator_c`

https://github.com/ros2/rosidl/blob/08617387920b40b8aa6b48db90d3535090be6086/rosidl_generator_c/resource/msg__type_support.h.em#L12-L16

https://github.com/ros2/rosidl/blob/master/rosidl_generator_c/resource/rosidl_generator_c__visibility_control.h.in

See how it's included in the `detail/interface__type_support.h` header generated by `rosidl_generator_c`.

https://github.com/ros2/rosidl/blob/08617387920b40b8aa6b48db90d3535090be6086/rosidl_generator_c/resource/idl__type_support.h.em#L28

Currently this generator has to set both compiler definitions because the `.cpp` generated by `rosidl_typesupport_c` file includes that typesupport header generated by `rosidl_generator_c`. If it didn't, the compiler would complain about the header saying dllimport while the .cpp file says dllexport. 

https://github.com/ros2/rosidl_typesupport/blob/97bc0a7716f7346dc0a34eeaba2e9b658f59beb5/rosidl_typesupport_c/cmake/rosidl_typesupport_c_generate_interfaces.cmake#L103-L106

This PR cleans it up by removing the visibility header generated by `rosidl_typesupport_c`. The header generated by `rosidl_generator_c` already has the symbol with an export macro, and the `.cpp` file includes it, so the `.cpp` file doesn't need it again. The result is it uses the visibility header generated by `rosidl_generator_c`.

Reply:
This had a missing piece. I assumed the `rosidl_generator_c` generated visibility header was included when building the `.cpp` file, but that wasn't the case. 57c6c7a fixes that.